### DISCUSSION TITLE
How to use variants with @stitches/core

### DISCUSSION THREAD

**Discussion by dartilesm** [stitchesjs/stitches#939] [Help] [answered]
Hi everyone, I'm trying stitches in a vue project and have no idea how to use variants in this framework. With react, variants it's used as props because we use `styled` function to create components directly, but whit `@stitches/core` I think we can not create components with this functions.

The point is, I'm using `css` function to create styles and pass it as class to my components and I dont know how to get adventage with variants.

Let you an example in codesanbox
https://codesandbox.io/s/stitches-in-vue-forked-vektj?file=/src/components/Button.styles.js

I am spanish speaker, sorry for my bad english 🙈

**Comment by bwyx** [comment, marked as answer]
`css()` also accepts `variants` props, you can use that to create variants just like the react one, then pass the component props to the returned function.

here's a working example https://codesandbox.io/s/stitches-with-vue-b7wvy8?file=/src/styles/text.style.js

hope this help

**Reply by dartilesm** [reply]
Thanks @bwyx !! It's very helpful to me
